rotkehlchen/db/schema.py

```diff
@@ -659,6 +659,20 @@
 );
 """
 
+DB_CREATE_ACCOUNTING_RULE = """
+CREATE TABLE IF NOT EXISTS accounting_rules(
+    type TEXT NOT NULL,
+    subtype TEXT NOT NULL,
+    counterparty TEXT NOT NULL,
+    taxable INTEGER NOT NULL CHECK (taxable IN (0, 1)),
+    count_entire_amount_spend INTEGER NOT NULL CHECK (count_entire_amount_spend IN (0, 1)),
+    count_cost_basis_pnl INTEGER NOT NULL CHECK (count_cost_basis_pnl IN (0, 1)),
+    method TEXT,
+    accounting_treatment TEXT,
+    PRIMARY KEY(type, subtype, counterparty)
```

Comment:
This is a bad idea. And you can see it that you are actually trying to work with rowids, so exposing implementation details into our logic. Very bad.

If you need to work with a single identifier, and you seem to want to then make the schema explicit. This is what happens underneath so long as we don't use this `NOROWID` thing.

```suggestion
    identifier INTEGER NOT NULL PRIMARY KEY,
    type TEXT NOT NULL,
    subtype TEXT NOT NULL,
    counterparty TEXT NOT NULL,
    taxable INTEGER NOT NULL CHECK (taxable IN (0, 1)),
    count_entire_amount_spend INTEGER NOT NULL CHECK (count_entire_amount_spend IN (0, 1)),
    count_cost_basis_pnl INTEGER NOT NULL CHECK (count_cost_basis_pnl IN (0, 1)),
    method TEXT,
    accounting_treatment TEXT,
    UNIQUE(type, subtype, counterparty)
```

This is also suggested by their docs: https://www.sqlite.org/rowidtable.html


![2023-10-07_00-40](https://github.com/rotki/rotki/assets/1658405/857de232-ca9c-47a6-beef-815616921d57)